flake.lock

```diff
@@ -0,0 +1,185 @@
+{
+  "nodes": {
+    "common-infra": {
+      "inputs": {
+        "deploy": "deploy",
+        "nixpkgs": "nixpkgs"
+      },
+      "locked": {
+        "lastModified": 1612203367,
+        "narHash": "sha256-z4knCUz+FJxqEgbT+EEFuzpAFWHmWBfAfTr3HuHMKE0=",
+        "owner": "serokell",
+        "repo": "common-infra",
+        "rev": "c5b609969578f8f5ca56c7d439eb010fcbe17157",
+        "type": "github"
+      },
+      "original": {
+        "owner": "serokell",
+        "repo": "common-infra",
+        "type": "github"
+      }
+    },
+    "deploy": {
+      "inputs": {
+        "flake-compat": "flake-compat",
+        "naersk": "naersk",
+        "nixpkgs": [
+          "nixpkgs"
+        ],
+        "utils": "utils"
+      },
+      "locked": {
+        "lastModified": 1608488342,
+        "narHash": "sha256-JhEoKKmETJsdgS+kjT1rACnlM50q/JdVNyAfIbbVQcc=",
+        "owner": "serokell",
+        "repo": "deploy-rs",
+        "rev": "91e934fd9e1e06044f75a3e3ad998813c2a90fb5",
+        "type": "github"
+      },
+      "original": {
+        "owner": "serokell",
+        "repo": "deploy-rs",
+        "type": "github"
+      }
+    },
+    "flake-compat": {
+      "flake": false,
+      "locked": {
+        "lastModified": 1600853454,
+        "narHash": "sha256-EgsgbcJNZ9AQLVhjhfiegGjLbO+StBY9hfKsCwc8Hw8=",
+        "owner": "edolstra",
+        "repo": "flake-compat",
+        "rev": "94cf59784c73ecec461eaa291918eff0bfb538ac",
+        "type": "github"
+      },
+      "original": {
+        "owner": "edolstra",
+        "repo": "flake-compat",
+        "type": "github"
+      }
+    },
+    "hackage": {
+      "flake": false,
+      "locked": {
+        "lastModified": 1613437861,
+        "narHash": "sha256-fhUWF2dKVoPUyfuj7nj6nOcmu1EIVJ2BEEOyp2YdF3U=",
+        "owner": "input-output-hk",
+        "repo": "hackage.nix",
+        "rev": "03b3577a069b61a5fd6f9319e09dd67ad3d94aee",
+        "type": "github"
+      },
+      "original": {
+        "owner": "input-output-hk",
+        "repo": "hackage.nix",
+        "type": "github"
+      }
+    },
+    "haskell-nix": {
+      "locked": {
+        "lastModified": 1613524507,
+        "narHash": "sha256-vIjgOb6PKGtLpSDbKV6J2dI4nR8SOcST1I1B+qOhsR8=",
+        "owner": "input-output-hk",
+        "repo": "haskell.nix",
+        "rev": "ba87ffb84bc4158ada14908bddee9df2c81ac31b",
+        "type": "github"
+      },
+      "original": {
+        "owner": "input-output-hk",
+        "repo": "haskell.nix",
+        "type": "github"
+      }
+    },
+    "naersk": {
```

Comment:
https://github.com/nmattia/naersk says "Nix support for building cargo crates". Do we need it?

Reply:
It's used by `deploy-rs`. `flake.lock` contains all transitive dependencies for all inputs in the flake. `deploy-rs` is a transitive dependency of `common-infra` which is an input to this flake.

Reply:
AFAIU we don't use `deploy-rs` either, but I see that it's a dependency of `common-infra`, so ok. It's a bit poor granularity (that we need to mention so many packages just for a small piece of `common-infra`), but I guess we will end up using `deploy-rs` anyway, so it's not a problem.

Reply:
`common-infra` has 2 nix functions: `mkNode` to define deployment nodes, and `mkPipelineFile` that generates CI steps which uses the former to generate deployment steps. So we're technically using half of `common-infra` :stuck_out_tongue: 

Packages mentioned in `flake.lock` don't necessarily end up in any of the closures, and since Nix is lazy, they don't even get evaluated unless you actually use them somewhere. The lockfile is eager and will contain all transitive locks regardless of whether they are used. It's just extra noise, that's all.